public/app/features/explore/RichHistory/RichHistoryAddToLibrary.tsx

```diff
@@ -48,12 +27,14 @@ export const RichHistoryAddToLibrary = ({ query }: Props) => {
         isOpen={isOpen}
         onDismiss={() => setIsOpen(false)}
       >
-        <RichHistoryAddToLibraryForm
+        <AddToLibraryForm
           onCancel={() => setIsOpen(() => false)}
           query={query}
-          onSave={(data) => {
-            submit(data);
-            setIsOpen(false);
+          onSave={(isSuccess) => {
+            if (isSuccess) {
+              setIsOpen(false);
+              setHasBeenSaved(true);
```

Comment:
Is it needed now if `isSuccess` is handed to manage `isOpen`? Could we just use `isOpen` to control the modal? 🤔

Reply:
> Is it needed now if `isSuccess` is handed to manage `isOpen`? Could we just use `isOpen` to control the modal? 🤔

I don't understand what's redundant, here's what the three variables here do..

- `isSuccess` comes back from the save function to indicate the save was successful and was not errored
- `isOpen` indicates if the modal for that instance of the button is open or not. 
- `hasBeenSaved` indicates if the query for that instance of the button has been saved already and if it should show up. It doesn't persist but gives the user feedback that the entry exists in the other tab.

So `isOpen` can be false and `hasBeenSaved` can be true or false - if the save was cancelled or not or was never tried.